catboost/spark/catboost4j-spark/core/src/main/scala/ai/catboost/spark/impl/Workers.scala

```diff
@@ -126,6 +127,8 @@ private[spark] class CatBoostWorker(partitionId : Int) extends Logging {
         ()
       )
 
+      service.shutdown()
```

Comment:
- That's too early, the futures might not even have started to execute.
- In case of exception in the code of futures creation and submission this code won't be called.

It's appropriate to move the `try {` block opening below higher just after the creation of `service` / `ecsPool` and put `service`/`ecsPool`.shutdown() in the corresponding `finally` block, similar to how it is done in CatBoostPredictor.scala.